### PR TITLE
DAOS-3506 smd: SMD unit test failure fix

### PR DESCRIPTION
**Pull request by vvenkates27**
Adding missing ABT_init for SMD test failure. This is a unit test
issue. Also adding smd_ut to run_tests, looks like this was removed.

Signed-off-by: Vishwanath Venkatesan <vishwanath.venkatesan@intel.com>

### CODE REVIEW

File: src/bio/smd/tests/smd_ut.c [hunk 1]
```diff
@@ -251,6 +251,15 @@ static const struct CMUnitTest smd_uts[] = {
 
 int main(int argc, char **argv)
 {
+	int	rc;
+
+	rc = ABT_init(0, NULL);
```
Comment (jolivier23) [inline src/bio/smd/tests/smd_ut.c]: Shouldn't there be an ABT_finalize() call?

Comment (vvenkates27) [reply src/bio/smd/tests/smd_ut.c]: indeed it should be there, I will update this patch, once I figure out why run_tests.sh is failing in CI.

File: Jenkinsfile [hunk 2]
```diff
@@ -460,6 +460,7 @@ pipeline {
                                                  build/src/security/tests/srv_acl_tests,
                                                  build/src/vos/vea/tests/vea_ut,
                                                  build/src/common/tests/umem_test,
+												 build/src/bio/smd/tests/smd_ut,
```
Comment (jolivier23) [inline Jenkinsfile]: are we mixing tabs and spaces here or some such?   alignment is off

Comment (vvenkates27) [reply Jenkinsfile]: spaces and tabs are not treated properly.

**Pull request merged**